src/argilla/server/alembic/env.py

```diff
@@ -68,27 +70,31 @@ def run_migrations_offline() -> None:
         context.run_migrations()
 
 
-def run_migrations_online() -> None:
+def apply_migrations(connection) -> None:
+    context.configure(connection=connection, target_metadata=target_metadata)
+
+    with context.begin_transaction():
+        context.run_migrations()
+
+
+async def run_migrations_online() -> None:
     """Run migrations in 'online' mode.
 
     In this scenario we need to create an Engine
     and associate a connection with the context.
 
     """
-    connectable = engine_from_config(
+    connectable = async_engine_from_config(
         config.get_section(config.config_ini_section),
         prefix="sqlalchemy.",
         poolclass=pool.NullPool,
     )
 
-    with connectable.connect() as connection:
-        context.configure(connection=connection, target_metadata=target_metadata)
-
-        with context.begin_transaction():
-            context.run_migrations()
+    async with connectable.connect() as connection:
+        await connection.run_sync(apply_migrations)
 
 
 if context.is_offline_mode():
     run_migrations_offline()
```

Comment:
this line should also be wrapped in an async statement, right?

Reply:
yeah, I haven't updated it as we're not using the offline migrations, but I'll do it just in case.

Reply:
but now `run_migrations_offline` is a coroutine so the main program can stop before finishing the coroutine completion